apps/tx-builder/src/components/forms/SolidityForm.tsx

```diff
@@ -0,0 +1,202 @@
+import { useState } from 'react';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { ButtonLink, Switch, Text } from '@gnosis.pm/safe-react-components';
+import styled from 'styled-components';
+
+import { ContractInterface } from '../../hooks/useServices/interfaceRepository';
+import {
+  ADDRESS_FIELD_TYPE,
+  AMOUNT_FIELD_TYPE,
+  CONTRACT_METHOD_FIELD_TYPE,
+  HEX_ENCODED_DATA_FIELD_TYPE,
+} from './fields/fields';
+import Field from './fields/Field';
+import { Examples } from '../Examples';
+import { encodeToHexData } from '../../utils';
+
+function SolidityForm({
+  id,
+  onSubmit,
+  getAddressFromDomain,
+  initialValues,
+  nativeCurrencySymbol,
+  networkPrefix,
+  contract,
+  children,
+}: SolidityFormPropsTypes) {
+  const [showExamples, setShowExamples] = useState<boolean>(false);
+  const [showHexEncodedData, setShowHexEncodedData] = useState<boolean>(false);
+
+  const { handleSubmit, control, setValue, watch, getValues, reset, clearErrors } = useForm<SolodityFormValues>({
+    defaultValues: initialValues,
+    mode: 'onTouched', // This option allows you to configure the validation strategy before the user submits the form
```

Comment:
Good catch!
From [the docs](https://react-hook-form.com/api/useform#main):

- mode: `all`:  Validation will trigger on the blur and change events.
- mode: `onTouched`:  Validation will trigger on the first blur event. After that, it will trigger on every change event.

Basically, I used `onTouched` mode because when you start typing, the error doesn't show until you blur the field for first time (after this, validation is triggered on every change field event)

But I will update this to mode: `all` because its the current behaviour in the tx-builder.

Reply:
It's up to you. The comment seemed as though it was copied from an example.

Reply:
For me personally onTouched leads to a better UX @DaniSomoza

Reply:
restored mode to `onTouched`